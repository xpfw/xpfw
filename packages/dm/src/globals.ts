const StatType = {
  sum: "sum",
  mean: "mean",
  timeStep: "timeStep",
  sumTimeDistance: "sumTimeDistance",
  sumPrevTimeDistance: "sumPrevTimeDistance",
  avgTimeDistance: "avgTimeDistance",
  avgPrevTimeDistance: "avgPrevTimeDistance"
}

export {
  StatType
}
