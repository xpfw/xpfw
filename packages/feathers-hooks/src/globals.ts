const Method = {
  Create: "create",
  Update: "update",
  Patch: "patch",
  Find: "find",
  Remove: "remove",
  Get: "get"
}

export {
  Method
}
