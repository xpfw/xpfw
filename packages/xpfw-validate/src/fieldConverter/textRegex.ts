import { FieldConverter } from "../typeDef"

const convertTextToMongoRegex: FieldConverter = (value) => {
    return {
        $regex: `(.*?)${value}(.*?)`,
        $options: "isg"
    }
}

export default convertTextToMongoRegex
