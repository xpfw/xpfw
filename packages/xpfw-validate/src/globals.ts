const RequiredType = {
  Always: 2,
  AllNotSet: 3,
  SetToVal: 4
}

const ErrorType = {
  invalidType: 3,
  required: 4,
  tooLow: 5,
  tooHigh: 420,
  notEqual: 6,
  emptyUpdate: 7
}

const FieldType = {
  Text: 2,
  Number: 3,
  Boolean: 4,
  Date: 5,
  Select: 6,
  Password: 7,
  Array: 10,
  Object: 12,
  RelationshipSingle: 13,
  RelationshipMulti: 14,
  Slider: 15,
  Location: 16
}

const Permission = {
    Guest: "guest",
    User: "user",
    Owner: "owner",
    Admin: "admin",
    Server: "server"
}

const Method = {
  Create: "create",
  Update: "update",
  Patch: "patch",
  Find: "find",
  Remove: "remove",
  Get: "get"
}

const StatType = {
  sum: "sum",
  mean: "mean",
  timeStep: "timeStep",
  sumTimeDistance: "sumTimeDistance",
  sumPrevTimeDistance: "sumPrevTimeDistance",
  avgTimeDistance: "avgTimeDistance",
  avgPrevTimeDistance: "avgPrevTimeDistance"
}

const DateType = {
  dateTime: 2,
  date: 3,
  time: 4
}

const options = {
  userCollection: "users",
  userIdPath: "_id"
}

export {
  StatType, RequiredType, ErrorType, FieldType, Permission, Method, DateType, options
}
export default {
  StatType, RequiredType, ErrorType, FieldType, Permission, Method, DateType, options
}
