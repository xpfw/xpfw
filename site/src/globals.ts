import { StatType } from "@xpfw/dm"
import { executeForMethods, ExtendedJSONSchema } from "@xpfw/form"
import { isString } from "lodash"
import * as momentA from "moment"
import { changeValToRegex } from "./components/ui/regex"
const moment: any = momentA

const TagName: ExtendedJSONSchema = {
  title: "tagName",
  type: "string"
}
const TagDescription: ExtendedJSONSchema = {
  title: "description",
  type: "string"
}

const TagModel: ExtendedJSONSchema = {
  title: "tagModel",
  collection: "tags",
  required: [String(TagName.title)],
  properties: {
    [String(TagName.title)]: TagName,
    [String(TagDescription.title)]: TagDescription
  }
}

const Title: ExtendedJSONSchema = {
  title: "title",
  type: "string"
}

const CreatedAt: ExtendedJSONSchema = {
  title: "createdAt",
  type: "string",
  format: "date-time"
}

const Tags: ExtendedJSONSchema = {
  title: "tags",
  type: "array",
  theme: "multi",
  relationship: {
    namePath: TagName.title,
    collection: TagModel.collection,
    idPath: "_id"
  },
  modify: changeValToRegex(String(TagName.title))
}

const TagCollectionStats = [{
  id: "Some",
  type: StatType.sum,
  options: {itemPath: `${Tags.title}.length`}
}, {
  id: "Average",
  type: StatType.mean,
  options: {itemPath: `${Tags.title}.length`}
}, {
  id: "timeDistance",
  type: StatType.avgPrevTimeDistance,
  options: {itemPath: CreatedAt.title}
}, {
  id: "timeStepson",
  type: StatType.timeStep,
  options: {subType: StatType.sum, subConfig: {itemPath: `${Tags.title}.length`}}
}]

const ensureDate = (path: string, methods?: string[]) => {
  return executeForMethods((value, schema, method) => {
    if (isString(value[path])) {
      value[path] = moment(value[path]).toDate()
    }
    return Promise.resolve(value)
  }, methods)
}

const TagCollectionModel: ExtendedJSONSchema = {
  title: "tagColModel",
  collection: "tagCol",
  required: [String(Title.title)],
  properties: {
    [String(Title.title)]: Title,
    [String(Tags.title)]: Tags,
    [String(CreatedAt.title)]: CreatedAt
  },
  modify: [ensureDate(String(CreatedAt.title), ["create", "update"])]
}

const RecipeName: ExtendedJSONSchema = {
  title: "name",
  type: "string"
}

const RecipeAuthor: ExtendedJSONSchema = {
  title: "createdBy",
  type: "string"
}

const RecipeDate: ExtendedJSONSchema = {
  title: "createdAt",
  type: "object",
  format: "date-time"
}

const RecipeModel: ExtendedJSONSchema = {
  title: "recipeModel",
  collection: "recipes",
  required: [String(RecipeName.title), String(RecipeAuthor.title)],
  properties: {
    [String(RecipeName.title)]: RecipeName,
    [String(RecipeAuthor.title)]: RecipeAuthor
  }
}

const siteGlobals = {
  linkClass: "tag is-rounded is-white has-text-link is-size-6",
  contentClass: "is-size-5 smallMarginTop has-text-centered",
  externalLinkConfig: {
    className: "has-text-link",
    target: "_blank"
  },
  gitRoot: "https://github.com/xpfw/xpfw/",
  pkgRoot: "https://github.com/xpfw/xpfw/tree/master/packages/"
}

export default siteGlobals
export {
  TagModel, TagCollectionModel, TagName, TagDescription, Title, Tags,
  RecipeModel, RecipeName, RecipeAuthor, RecipeDate, CreatedAt, TagCollectionStats
}
