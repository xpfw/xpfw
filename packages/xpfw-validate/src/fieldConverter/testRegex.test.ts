import { FieldType } from "../globals"
import textRegex from "./textRegex"

it("Expect createFormData behaves consistently", () => {
    expect(textRegex("myString")).toMatchSnapshot("result of to regex")
})
