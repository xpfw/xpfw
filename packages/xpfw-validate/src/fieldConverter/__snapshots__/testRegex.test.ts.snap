// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Expect createFormData behaves consistently: result of to regex 1`] = `
Object {
  "$options": "isg",
  "$regex": "(.*?)myString(.*?)",
}
`;
